master/internal/spec_util_intg_test.go

```diff
@@ -0,0 +1,133 @@
+//go:build integration
+// +build integration
+
+package internal
+
+import (
+	"testing"
+
+	"github.com/determined-ai/determined/master/internal/config"
+	"github.com/determined-ai/determined/master/internal/mocks"
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/archive"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/tasks"
+	"github.com/determined-ai/determined/proto/pkg/utilv1"
+	"github.com/stretchr/testify/require"
+	k8sV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func getMockResourceManager(poolName string) *mocks.ResourceManager {
+	rm := &mocks.ResourceManager{}
+	rm.On("ResolveResourcePool", "/", 0, 1).Return(poolName, nil)
+	rm.On("ValidateResourcePoolAvailability", &sproto.ValidateResourcePoolAvailabilityRequest{
+		Name:  poolName,
+		Slots: 1,
+	}).Return(nil, nil)
+	rm.On("ValidateResources", poolName, 1, true).Return(nil)
+	return rm
+}
+
+func TestResolveResources(t *testing.T) {
```

Comment:
all these tests are the sort where setting up table driven tests right now will help the next person around a lot and allow you to cover a lot more cases quickly, now.